ush/extractvars_tools.sh

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env bash
+
+gen_parmlist() {
+  infileg=$1
+  requestedvar_in_file1=$2
+  varlist=$3
+  file_vars=$(${WGRIB2} "${infileg}")
+  while read -r vari; do
+    if [[ "${file_vars}" =~ ${vari} && -n "${vari}" ]]; then
+      echo "${vari}" >> "${requestedvar_in_file1}"
+    fi
+  done <"${varlist}"
+}
+
+check_atmos() {
+  requestedvar_in_file1=$1
+  requestedvar_in_file2=$2
+  varlistl=$3
+  fnhl=$4
+  requestedvar_in_allgrb2file="${subdata}/requestedvar_in_allgrb2file.txt"
+  rm -rvf "${requestedvar_in_allgrb2file}"
+  cat "${requestedvar_in_file1}" "${requestedvar_in_file2}" >> "${requestedvar_in_allgrb2file}"
+  mapfile -t requestedvar_in_allgrb2file_arr < "${requestedvar_in_allgrb2file}"
+  while read -r vari; do
+    if [[ ! ${requestedvar_in_allgrb2file_arr[*]} =~ ${vari} ]] ;then
+      echo "WARNING: PARM VARIABLE (${vari}) is not available in pgrb and pgrb2b for f${fnhl}."
+    fi
+  done <"${varlistl}"
+}
+
+daily_avg_atmos() {
+  outfile_p=$1
+  fnd=$(printf "%2.2d" "${dcnt}")
+  davg_file=${outdirpre}/${RUN}.${cycle}.pgrb2.${outres}.24hr_avg.ldy${fnd}
+  vcnt=1
+  while read -r vari; do
+    davgtmp=${subdata}/${RUN}.${cycle}.tmp.pgrb2.${outres}.ldy${fnd}.${vcnt}
+    # shellcheck disable=SC2312
+    ${WGRIB2} "${outfile_p}" | grep "${vari}" | ${WGRIB2} -i "${outfile_p}" -fcst_ave 6hr "${davgtmp}"
+    # shellcheck disable=SC2312
+    ${WGRIB2} "${davgtmp}" | ${WGRIB2} -i "${davgtmp}" -append -grib "${davg_file}"
+    rm -f "${davgtmp}"
+    vcnt=$(( vcnt + 1 ))
+  done <"${varlist_d}" #variable
+}
+
+copy_to_comout() {
+  rundir_outfile=$1 #output data file generated in RUNDIR
+  comout_dir=$2 #destination directory to which to copy the data file
+  if [[ -f "${rundir_outfile}" ]];then
+    cp -pr "${rundir_outfile}" "${comout_dir}"
+  else
+   echo "WARNING: Output variable (${rundir_outfile}) does not exist."
+  fi
+}
```

Comment:
I think these functions need to be exported to be able to be sourced in `exglobal_extractvars.sh`.

e.g.
```suggestion
}

export -f gen_parmlist
export -f check_atmos
export -f daily_avg_atmos
export -f copy_to_comout
```

Doing so will allow removing the repeated sourcing in the `ush/` scripts

Reply:
Recommend using `declare -xf` instead for consistency with `bash_utils.sh` (both methods do the same thing).